private/syntax-replacement.rkt

```diff
@@ -295,6 +301,35 @@
        (check-equal? (syntax-replacement-render replacement) expected)])))
 
 
+(define (syntax-replacement-preserves-free-identifiers? replacement)
+  (match replacement
+    [(syntax-replacement #:original-syntax orig
+                         #:new-syntax new
+                         #:introduction-scope intro)
+     ((stx-free-ids=?/ignore
+       (ignore-id? intro (list #'SPACE #'NEWLINE #'ORIGINAL-GAP #'ORIGINAL-SPLICE)))
+      new
+      (datum->syntax orig (syntax->datum new)))]))
```

Comment:
How does this work, exactly? You're checking that all of the identifiers are `free-identifier=?` to themselves when they have all their scopes removed and replaced with just the scopes from the original syntax object, right?